sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/EdmType.java

```diff
@@ -0,0 +1,214 @@
+
+    /**
+     * <strong>Edm.Decimal</strong> Represents numeric values with fixed precision and scale. This type can describe a
+     * numeric value ranging from negative 10^255 + 1 to positive 10^255 -1
+     */
+    DECIMAL,
+
+    /**
+     * <strong>Edm.Double</strong> Represents a floating point number with 15 digits precision that can represent values
+     * with approximate range of +/- 2.23e -308 through +/- 1.79e +308
+     */
+    DOUBLE,
+
+    /**
+     * <strong>Edm.Single</strong> Represents a floating point number with 7 digits precision that can represent values
+     * with approximate range of +/- 1.18e -38 through +/- 3.40e +38
+     */
+    SINGLE,
+
+    /**
+     * <strong>Edm.Guid</strong> Represents a 16-byte (128-bit) unique identifier value
+     */
+    GUID,
+
+    /**
+     * <strong>Edm.Int16</strong> Represents a signed 16-bit integer value
+     */
+    INT16,
+
+    /**
+     * <strong>Edm.Int32</strong> Represents a signed 32-bit integer value
+     */
+    INT32,
+
+    /**
+     * <strong>Edm.Int64</strong> Represents a signed 64-bit integer value
+     */
+    INT64,
+
+    /**
+     * <strong>Edm.SByte</strong> Represents a signed 8-bit integer value
+     */
+    SBYTE,
+
+    /**
+     * <strong>Edm.String</strong> Represents fixed- or variable-length character data
+     */
+    STRING,
+
+    /**
+     * <strong>Edm.Time</strong> Represents the time of day with values ranging from 0:00:00.x to 23:59:59.y, where x
+     * and y depend upon the precision
+     */
+    TIME;
+
+    private static final Set<EdmType> UNANNOTATED = Collections.unmodifiableSet(EnumSet.of(BOOLEAN, DOUBLE, INT32,
+            STRING));
+
+    protected final boolean mustAnnotateType() {
+        return !UNANNOTATED.contains(this);
+    }
+
+    /**
+     * Parses an EDM data type name and return the matching {@link EdmType} enumeration value. A <code>null</code> or
+     * empty value parameter is matched as {@link EdmType#STRING}. Note that only the subset of EDM data types
+     * supported in Microsoft Azure Table storage is parsed, consisting of {@link #BINARY}, {@link #BOOLEAN},
+     * {@link #BYTE} , {@link #DATE_TIME}, {@link #DOUBLE}, {@link #GUID}, {@link #INT32}, {@link #INT64}, and
+     * {@link #STRING}. Any
+     * other type will cause an {@link IllegalArgumentException} to be thrown.
+     *
+     * @param value
+     *            A <code>String</code> containing the name of an EDM data type.
+     * @return
+     *         The {@link EdmType} enumeration value matching the specified EDM data type.
+     *
+     * @throws IllegalArgumentException
+     *             if an EDM data type not supported in Microsoft Azure Table storage is passed as an argument.
+     *
+     */
+    public static EdmType parse(final String value) {
+        if (value == null || value.length() == 0) {
+            return EdmType.STRING;
+        }
+        else if (value.equals(ODataConstants.EDMTYPE_DATETIME)) {
```

Comment:
This isn't the correct formatting.  The method name, should be `valueOf` to adhere to Java conventions.